src/vcpkg/tools.cpp

```diff
@@ -62,82 +62,444 @@ namespace vcpkg
+                "6403dac9196baffcff614fa73ea530752997c8db6bbfbaa0446b4b09d7327e2aa6e8615d1283c961d3bf0df497e85ba8660414"
+                "9f1505ee75f89d600245a45dde",
+                "ninja-linux-1.11.1.zip",
+            },
+            {
+                "ninja",
+                "osx",
+                nullopt,
+                "1.11.1",
+                "ninja",
+                "https://github.com/ninja-build/ninja/releases/download/v1.11.1/ninja-mac.zip",
+                "dad33b0918c60bbf5107951a936175b1610b4894a408f4ba4b47a2f5b328fc982a52a2aed6a0cb75028ee4765af5083bea6661"
+                "1c37516826eb0c851366bb4427",
+                "ninja-mac-1.11.1.zip",
+            },
+            {
+                "powershell-core",
+                "windows",
+                nullopt,
+                "7.2.23",
+                "pwsh.exe",
+                "https://github.com/PowerShell/PowerShell/releases/download/v7.2.23/PowerShell-7.2.23-win-x64.zip",
+                "b374a878df02980d54e17ad7cfc9021e331748c3770f586be61356c257494d1b33899c1167d09a35c210bc084474aefdff972f"
+                "672d16afb43be0562b3589285a",
+                "PowerShell-7.2.23-win-x64.zip",
+            },
+            {
+                "node",
+                "windows",
+                nullopt,
+                "16.15.1",
+                "node-v16.15.1-win-x64/node.exe",
+                "https://nodejs.org/dist/v16.15.1/node-v16.15.1-win-x64.7z",
+                "7ec4bfe2ea6034e1461e306b6372d62c0c5d1060c453ba76a73a5cec38ac26b5952a744caa9071455329caa58eb0a96d26c688"
+                "54c8915c17610ff27b0cf2c1cf",
+                "node-v16.15.1-win-x64.7z",
+            },
+            {
+                "node",
+                "linux",
+                nullopt,
+                "16.15.1",
+                "node-v16.15.1-linux-x64/bin/node",
+                "https://nodejs.org/dist/v16.15.1/node-v16.15.1-linux-x64.tar.gz",
+                "5ad3b4b9caeaa8d31503efa99f5a593118a267dec9d4181d019732126ba248ce9a901207115b3f6b899eb5b3f0373c7f77ea95"
+                "cc92ac625cddf437ee9b8b8919",
+                "node-v16.15.1-linux-x64.tar.gz",
+            },
+            {
+                "node",
+                "osx",
+                nullopt,
+                "16.15.1",
+                "node-v16.15.1-darwin-x64/bin/node",
+                "https://nodejs.org/dist/v16.15.1/node-v16.15.1-darwin-x64.tar.gz",
+                "90d0612bbe5467b6cf385c91a68b8daad0057e3e0ccacea44567f5b95b14f7481cb79784185ab1463b4bd990e092ff0f910957"
+                "6d1a1934b84e1c816582929611",
+                "node-v16.15.1-darwin-x64.tar.gz",
+            }};
+
+        size_t default_tool = 0;
+        bool existsDefaultTool = false;
+        for (std::size_t i = 0; i < tool_data.size(); i++)
+        {
+            ArchToolData& d = tool_data[i];
+            if (d.tool == toolname && d.arch.has_value() && d.arch.value_or_exit(VCPKG_LINE_INFO) == arch && d.os == os)
```

Comment:
This loop body is not clear to me, can you comment what it's trying to do?

Reply:
If we can find a tool with a matching tool name, architecture, and platform, use that.

Otherwise, choose the tool with a matching name and platform, and `nullopt` architecture, this is the default tool.

Reply:
Sorry, I should have been more clear: The code should be making that clear :)

Reply:
https://github.com/data-queue/vcpkg-tool/pull/1 I think this could be a lot clearer, I submitted this PR which hopefully makes sense.

However I realize I broke some tests, give me a moment